src/user-details/user-details.controller.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Param,
+  Post,
+  Query,
+  UseGuards,
+} from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import {
+  ApiParam,
+  ApiBearerAuth,
+  ApiCreatedResponse,
+  ApiOkResponse,
+  ApiBody,
+  ApiUnauthorizedResponse,
+} from '@nestjs/swagger';
+import { GetUser } from 'src/auth/get-user.decorator';
+import { User } from 'src/auth/user.entity';
+import { CreateUserDetailsDto } from './Dto/create-user-details-dto';
+import { UserDetails } from './entity/user-details.entity';
+import { UserDetailsService } from './user-details.service';
+
+@Controller('user-details')
+@UseGuards(AuthGuard())
+@ApiBearerAuth('access-token')
+@ApiUnauthorizedResponse({ description: 'Unauthorized' })
+export class UserDetailsController {
+  constructor(private userDetailsService: UserDetailsService) {}
+
+  @Get('/getUserDetails')
+  @ApiOkResponse({ description: 'Get user details' })
+  getUserDetails(@GetUser() user: User): Promise<UserDetails> {
+    return this.userDetailsService.getUserDetails(user);
+  }
+  @Post('/createDetails')
```

Comment:
[Suggestion] We should have a free space between class methods.

Reply:
yes , but i did it to add the userDetails that is logged in , i didnt need for the idUser

Reply:
so i made it the user to add his details , or update his details